src/core/maprenderer/qgsmaprendererjob.cpp

```diff
@@ -649,7 +649,7 @@ std::vector<LayerRenderJob> QgsMapRendererJob::prepareJobs( QPainter *painter, Q
       job.context()->setElevationMap( job.elevationMap );
     }
 
-    if ( ( job.renderer->flags() & Qgis::MapLayerRendererFlag::RenderPartialOutputs ) && ( mSettings.flags() & Qgis::MapSettingsFlag::RenderPartialOutput ) )
+    if ( job.renderer && ( job.renderer->flags() & Qgis::MapLayerRendererFlag::RenderPartialOutputs ) && ( mSettings.flags() & Qgis::MapSettingsFlag::RenderPartialOutput ) )
```

Comment:
are this additional check and the one in qgsmaprenderedparalleljob.cpp specific to the new dev, or bug fixes ? If the later, they should probably be in a separate commit (apologies if it is already the case, but wasn't obvious from looking at commit titles)